cosmwasm/ibc-union/lightclient/state-lens-ics23-smt/src/errors.rs

```diff
@@ -47,7 +47,7 @@ pub enum Error {
     CounterpartyStorageNotNil,
 
     #[error("could not decode the proof: {0}")]
-    ProofDecode(Bytes),
+    ProofDecode(bincode::error::DecodeError),
```

Comment:
nit: this should use `#[from]` instead of printing the error directly

Reply:
apparently the decodeerror dont implement error hence from and source wont work